How-To-Use-NDB.md

```diff
@@ -0,0 +1,260 @@
+
+This document will take through a tour of using `ndb` and it's features. This document assumes you already understand the SQL language. If you are not familiar with SQL, I recommend visiting [this free CodeAcademy course](https://www.codecademy.com/learn/learn-sql) or searching for the many other resources online for learning SQL first.
+
+## CONNECTING
+
+When writing your nim program, you will want to first import the specific sub-library for your database.
+
+For SQLite:
+
+```nim
+import ndb / sqlite
+```
+
+For PostgreSQL:
+
+```nim
+import ndb / postgresql
+```
+
+For MySQL or MariaDB:
+
+```nim
+import ndb / mariadb
+```
+
+The remaining examples in this document assume you are working SQLite, but the other database types should behave mostly the same.
+
+Connect to the database using the `open` function and assign the result to a variable to hold the connection. And, later close it using the 'close' function.
+
+```nim
+import ndb / sqlite
+
+let db = open("example.db", "", "", "")
+
+# Add the database commands here.
+
+db.close()
+```
+
+**EXTRA:** [Behind the Scenes](How-To-Use-NDB-behind-the-scenes.md)
+
+**EXTRA:** [Connecting to Multiple Databases](How-To-Use-NDB-connecting-to-multiple-databases.md)
+
+## MAKING TABLES
+
+There are no specific functions for making new tables in `ndb`, instead you will simply rely on sending direct SQL statements to do it. That is done using the `exec` function.
+
+This function, `exec` is fairly powerful and can also do parameter substitution. We will see more of that the CREATING RECORDS section below.
+
+An example of making a new table:
+
+```nim
+import ndb / sqlite
+let db = open("example.db", "", "", "")
+
+let creation_query = """
+    CREATE TABLE myGarden (
+        id INTEGER PRIMARY KEY,
+        name STRING NOT NULL,
+        qty INTEGER,
+        price FLOAT,
+    )
+"""
+db.exec(creation_query)
+
+db.close()
+```
+
+## CREATING RECORDS
+
+You can, of course, simply create new records in a table by issueing a `exec` query to do so:
+
+```nim
+db.exec("""
+    INSERT INTO MyGarden (id, name, qty, price) VALUES (1, "carrot", 20, 0.33)
+""")
+db.exec("""
+    INSERT INTO MyGarden (id, name, qty, price) VALUES (2, "squash", 32, 0.5)
+""")
+```
+
+Or you can use `exec`'s parameter substitution:
+
+```nim
+db.exec("""
+    INSERT INTO MyGarden (id, name, qty, price) VALUES (?, ?, ?, ?)
+""", 3, "cucumber", 22, 1632.2)
+```
+
+But `ndb` also supports a special function called `tryInsertID` which not only inserts the row but also create a new unique number for a matching `id` column and returns the value of that new id (`int64`).
+
+```nim
+var new_id = db.insertID("""
+    INSERT INTO MyGarden (name, qty, price) VALUES (?, ?, ?)
+""", "green bean", 100, 0.92)
+```
+
+If an error is found during the insertion, the returned number is -1. If you are wanting to do your own error catching, then use the alternate `insertID` function. See [CATCHING ERRORS] .
```

Comment:
I would rather see it's the other way around.
`insertID` is the default that leverages nim's exceptions, and `tryInsertID` should be used for custom error handling in a case when exceptions are not convenient.